File: Baekjoon/Class2/덩치.py
n = int(input())
people = []
for _ in range(n):
    w, h = map(int, input().split())
    people.append((w, h))
    
ranks = []
for i in people:
    rank = 1
    for j in people:
        if (i[0] != j[0]) and (i[1] != j[1]):
            if (i[0] < j[0]) and (i[1] < j[1]):
                rank += 1
    ranks.append(rank)
for rank in ranks:
    print(rank, end=' ')

File: Baekjoon/Javascript/덩치.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

function solution(input) {
    [n, ...person] = input;
    let result = [];

    for (let i = 0; i < n; i += 1){
        let rank = 0;
        for (let j = 0; j < n; j += 1){
            if (i === j) continue;
            [weight1, tall1] = person[i].split(' ');
            [weight2, tall2] = person[j].split(' ');

            if (Number(weight1) < Number(weight2) && Number(tall1) < Number(tall2)) rank += 1;
        }
        result.push(rank + 1);
    }
    return result.join(' ');
}

console.log(solution(input));